### DISCUSSION TITLE
Feature Request: Show linked pull requests in Projects (beta)

### DISCUSSION THREAD

**Discussion by markdorison** [community/community#5003] [Projects and Issues] [answered]
Issues viewed on existing (non-beta) projects in board view show pull requests that have been linked to issues. Some form of this functionality would be fantastic in the new Projects (beta).

<img width="319" alt="CleanShot 2021-08-10 at 10 21 47@2x" src="https://user-images.githubusercontent.com/20355/128884111-54db987a-601b-4582-9e4f-7b02d23caa34.png">

**Comment by kevinwhoffman** [comment]
Agreed. When some issue in a project is ready for review, I want to have one-click access into the linked pull request so that I can quickly jump from the table view straight into review mode.

We currently have to click from the table view into the issue and then click through to the linked pull request.

Alternatively, the pull request itself could be added to the project, but in my experience this just creates more confusion. With issues and their linked pull requests in the same table view, the team has to juggle the status of both.

**Comment by rodsfreitas** [comment]
I didn't see this thread before opening a new one, but having the Linked PR would be really great. https://github.com/github/feedback/discussions/5535

**Comment by polevaultweb** [comment]
This would be very helpful - one click from an issue in a project table view to the PR for that issue:

![image](https://user-images.githubusercontent.com/1770201/135443988-b9cfaa91-acee-4c04-afdf-fa970bea715c.png)

**Comment by leanfrancucci** [comment]
In order to deal with this issue I've been thinking about using a table view that has a custom field (Text type) named PR. This field has links to pull requests that have been linked to issues :no_good:
![image](https://user-images.githubusercontent.com/33349769/139323969-8a44908e-2b6f-405e-ab3d-81ee54784afc.png)

**Comment by breity** [comment]
👍 Yes please!

As others have mentioned, it's very convenient to be able to jump directly to linked PRs from issues a project.

We also assign points to PRs with labels, so being able to view linked PR labels in the project view helps us quickly tally up total points when we do sprint retrospectives.

**Comment by zbjornson** [comment]
This is how our custom Kanban board shows PRs that contain `Fixes #xxx`:

![image](https://user-images.githubusercontent.com/469365/142089043-3fb21c8d-4cd0-499e-8085-bf16c954614b.png)

which was based on how Waffle.io used to show them:
![image](https://user-images.githubusercontent.com/469365/142089068-10222057-fafa-4655-b180-c0985b09a459.png)
*(screenshot from https://blog.oursky.com/2017/11/28/project-management-startup-founders/screen-shot-2017-06-12-at-3-06-36-pm/)*

I also like what @rodsfreitas showed in #5535.

**Comment by azenMatt** [comment, marked as answer]
Hi everyone - we just released this feature! 

You can check it out here - https://github.blog/changelog/2021-12-16-the-new-github-issues-12-16-update/

![changelog_-_linked_prs_v2 2c](https://user-images.githubusercontent.com/7584089/146577210-abb56ee5-37da-45eb-81a6-0b05f92fbb48.gif)

Let me know what you think! 

We have a couple more things come soon for this - including a reviewers field

**Reply by polevaultweb** [reply]
Awesome, thanks @azenMatt. A reviewer field will be much appreciated too!

**Reply by danielBingham** [reply]
The way it is shown now with a bubble is much less readable than the way it is currently shown on non-beta (old school) projects with a sub-card.  The way Waffle used to do it, with a subcard that could also show each PR's assignees (separate from the issue) and reviewers (again separate from the issue) was the perfect UX for this.  It allowed us to see everything we needed related to that issue and the PRs that contained its execution at a glance.  If you look at this comment (https://github.com/github/feedback/discussions/5003#discussioncomment-1655004) there's a screenshot.

**Comment by zbjornson** [comment]
@azenMatt that's a great start, but having two cards (one for the issue, one [or more] for the PR) clutters up the board. I really like the screenshots I posted just above (https://github.com/github/feedback/discussions/5003#discussioncomment-1655004), that markdorison posted in the OP and that rodsfreitas posted in https://github.com/github/feedback/discussions/5535, all of which nest PRs with their issues.

**Reply by azenMatt** [reply]
If you link the PR - but don't add it to the project - it should give you the same effect?

**Reply by kevinwhoffman** [reply]
> If you link the PR - but don't add it to the project - it should give you the same effect?

This is what we're doing and it's working well. But if you did need to include both issues and PRs in your project, you might find it useful to use an `is:issue` filter on your Kanban view. This will hide the PR cards while still showing them as linked PRs on the issue cards.

**Reply by zbjornson** [reply]
Similar; that's definitely viable. However, having a card can let you see more info like who the PR is assigned to, checks status, labels, milestone and review status.

*edit* Our PRs also don't always have a corresponding issue. We'd need some custom automation to be able to only add PRs if they don't have linked issues.

**Reply by zbjornson** [reply]
@azenMatt I edited my post above and wanted to make sure you saw the follow-up comments since I don't think a notification was sent with the edits.
- There's no way to see info about linked PRs, such as status/checks, labels, assignees, milestone, review status.
- The proposed workaround for avoiding two cards (only add issues and not PRs to the project) would require custom automation that still adds PRs that don't have corresponding issues.

Are these under consideration? Should I open a new feature request?

**Reply by azenMatt** [reply]
Great points @zbjornson - if you could open a new discussion for that it would make it easier to track.

**Reply by danielBingham** [reply]
@zbjornson Can you link that discussion here?  I'd love to vote it up, I have the same issues you do and would love to add my voice encouraging them to return to a more Waffle-like UX.

**Reply by zbjornson** [reply]
@danielBingham the "enhanced display of linked PRs" issue is https://github.com/github/feedback/discussions/10252.

**Reply by azenMatt** [reply]
thanks for taking the time to open that @zbjornson - i'l make sure it gets shared back with the right people

**Comment by paulschreiber** [comment]
@azenMatt this is working most of the time, but isn't working for a PR I have with a feature branch:
- issue: https://github.com/techmatters/terraso-web-client/issues/296
- PR: https://github.com/techmatters/terraso-web-client/pull/308

<img width="1384" alt="image" src="https://user-images.githubusercontent.com/86784/163829991-97150975-d7c0-4415-8a8a-0219c432b529.png">

**Reply by azenMatt** [reply]
Hi @paulschreiber - I don't see this linked in the issue: 

![image](https://user-images.githubusercontent.com/7584089/164420792-868055c1-ff78-475b-954e-652c1c8f39d6.png)

If you add the PR there, does this work for you?

**Reply by paulschreiber** [reply]
@azenMatt It's linked — I see it in your screen shot — right at the bottom — with the purple `Merged` beside it.

**Reply by kevinwhoffman** [reply]
@paulschreiber The issue shows that it was mentioned in the PR, but this is not the same as being linked. If the issue and pull request were linked, you would see that under the Development heading in the sidebar.

Even though you wrote `Fixes #296` in your PR description, the issue did not become linked because, as you mentioned, you are not targeting the default branch. This is by design according to the documentation for [Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

See the blue note at the top of the doc:

<img width="757" alt="Google Chrome-Linking a pull request to an issue - GitHub Docs-20220421T214854@2x" src="https://user-images.githubusercontent.com/1938671/164557587-00613f9d-a0ca-48cf-afbf-a410bf52f03b.png">

So it appears that the linked pull requests column is behaving as it should, and your feedback should probably be directed towards the linking feature in general if you think it should behave differently.

**Reply by paulschreiber** [reply]
@kevinwhoffman That's very helpful. It definitely makes sense that you don't _close_ the issue until the PR is merged in to main, but I would hope for the linking to be enough for it to show up in Projects.

**Comment by ygavenchuk** [comment]
I'd like to propose one more thing to improve here. I'm participating in an organization that has multiple repositories in it. One of those repositories is dedicated to routine stuff - documentation, issues, projects etc. Having such structure it's "slightly" inconvenient to add linked PRs to issues. So, I'm just curious if it's possible to add/link an **existing** PR from another repository to the issue?

**Reply by marcaaron** [reply]
+1000 would love this feature. We have an entirely separate repository where all our issues live and can't link any of those issues with PRs that are spread out across many repo.
